### PR TITLE
add `scc`

### PR DESCRIPTION
**Pull request by 0xdeafbeef**
I am not sure if I should add updated results to pr

### CODE REVIEW

**Comment by loyd** [comment]
I think, we should check both `HashIndex` and `HashMap`:
https://github.com/wvwwvwwv/conc-map-bench/tree/Intel?tab=readme-ov-file#results

But more important here is that (possibly) @xacrimon has some conflict of interest. Therefore, it is unclear whether he will accept this contribution. If not, I think we should fork this repository.

File: src/adapters/scc.rs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+impl<K, H> CollectionHandle for SccMapTable<K, H>
+where
+    K: Send + Sync + From<u64> + Copy + 'static + Hash + Eq + std::fmt::Debug,
+    H: BuildHasher + Default + Send + Sync + 'static + Clone,
+{
+    type Key = K;
+
+    fn get(&mut self, key: &Self::Key) -> bool {
+        self.0.get(key).is_some()
```
Comment (wvwwvwwv) [inline src/adapters/scc.rs]: Hi, I'm one of the authors of the `scc` crate, and just happened to see this PR :-) Note that `get` acquires exclusive ownership of the entry for _modification_, so it is not quite optimal here - I'd suggest you use `read` here.

Comment (0xdeafbeef) [reply src/adapters/scc.rs]: Resolved. Maybe get should be named `get_for_update` like in sql?

Comment (0xdeafbeef) [reply src/adapters/scc.rs]: ReadHeavy std throughput
ReadHeavy std throughput
@wvwwvwwv not a drammatic change

Comment (wvwwvwwv) [reply src/adapters/scc.rs]: This proves how correctly/meticulously the benchmark suite has been written: keys must be very uniformly distributed.

**Comment by xacrimon** [comment]
@loyd @0xdeafbeef I don't think I have any real conflict of interest here that matters to me, I don't exactly make money off it. Dashmap isn't the current winner in all tests, esp on m1 as of 5.5.3 and I'll happily show case this. I created this repo and spent all the time writing suites and collecting metrics to improve competition and innovation in this space. I'll look into merging this very soon.

**Comment by xacrimon** [comment]
This mostly means I now have an incentive to go work on new map designs again :)
Competition has been pretty stale the last year or two.

**Comment by wvwwvwwv** [comment]
Let the game begin @xacrimon :-)

**Pull request merged**